src/DependencyInjection/SonataUserExtension.php

```diff
@@ -178,7 +178,17 @@ public function configureGoogleAuthenticator($config, ContainerBuilder $containe
         }
 
         $container->setParameter('sonata.user.google.authenticator.forced_for_role', $config['google_authenticator']['forced_for_role']);
-        $container->setParameter('sonata.user.google.authenticator.ip_white_list', $config['google_authenticator']['ip_white_list']);
+
+        // NEXT_MAJOR: Remove this checks and only set the `trusted_ip_list`.
+        if (\array_key_exists('ip_white_list', $config['google_authenticator']) && \array_key_exists('trusted_ip_list', $config['google_authenticator'])) {
```

Comment:
`$config['google_authenticator']['ip_white_list']` and `$config['google_authenticator']['trusted_ip_list']` are always defined, there is no need to check with `array_key_exists`.

Given that, you could do something like:
```php
if (count($config['google_authenticator']['ip_white_list']) > 0 && $config['google_authenticator']['trusted_ip_list'] !== ['127.0.0.1']) {
    throw new \LogicException('Please use only ``trusted_ip_list`` parameter. ``ip_white_list`` is deprecated.');
}
```

The idea is that `trusted_ip_list` will always have `['127.0.0.1']` value since it's the default value specified in `Configuration`. 

And then you could do something like:

```php
$trustedIpList = $config['google_authenticator']['trusted_ip_list'];

if (\count($config['google_authenticator']['ip_white_list']) > 0) {
    $trustedIpList = $config['google_authenticator']['ip_white_list'];
}

$container->setParameter('sonata.user.google.authenticator.ip_white_list', $trustedIpList);
$container->setParameter('sonata.user.google.authenticator.trusted_ip_list', $trustedIpList);
```